src/_guides/language/sound-problems.md

```diff
@@ -466,29 +487,46 @@ filterValues((x) => (x [!as String!]).contains('Hello'));
 
 <hr>
 
-<a id="common-errors-and-warnings"></a>
-## Runtime errors
+### Incorrect type inference
 
-Dart enforces a sound type system. Roughly, this means
-you can't get into a situation where the value stored in a variable is
-different from the variable's static type. Like most modern statically
-typed languages, Dart accomplishes this with a combination of static
-(compile-time) and dynamic (runtime) checking.
+```nocode
+error - The return type '...' isn't a '...', as required by the closure's context - return_of_invalid_type_from_closure
+```
 
```

Comment:
Okay I ran the analyzer again with Dart 2.18 and 2.19 and I'm getting different errors. Perhaps I was using an older development version of Dart on that machine.

Since this has to do with type inference, it results in many different errors depending on the context, due to types flowing through the code in different ways according to each situation.

Perhaps it doesn't make a lot of sense to include a specific error due to this, and your explanation of "incorrect type inference" should be sufficient. What do you think?

```suggestion
```

Reply:
Definitely agree, especially now when we just need to get the update out for 2.18. Thanks so much!